network/delete.go

```diff
@@ -14,28 +18,66 @@ func Delete() error {
 
 	kubernetes := k8s.Kubernetes{Client: k8sClient, RestConfig: k8sRestConfig}
 
-	volumes, err := kubernetes.GetPVCs()
+	if !config.KeepLogsMetrics {
+		volumes, err := kubernetes.GetPVCs()
 
-	if err != nil {
-		return err
-	}
+		if err != nil {
+			return err
+		}
 
-	err = gcp.DeleteKubernetesCluster(volumes)
+		err = gcp.DeleteKubernetesCluster(volumes)
 
-	if err != nil {
-		return err
-	}
+		if err != nil {
+			return err
+		}
 
-	err = kubernetes.DeleteMetricsDNSRecords()
+		err = kubernetes.DeleteMetricsDNSRecords()
 
-	if err != nil {
-		return err
-	}
+		if err != nil {
+			return err
+		}
 
-	err = kubernetes.DeleteELKDNSRecords()
+		err = kubernetes.DeleteELKDNSRecords()
 
-	if err != nil {
-		return err
+		if err != nil {
+			return err
+		}
+	} else {
+		k8sRestConfig, k8sClient, err := gcp.GetKubernetesClient()
+
+		if err != nil {
+			return err
+		}
+
+		kubernetes := k8s.Kubernetes{Client: k8sClient, RestConfig: k8sRestConfig}
+
+		deployments, err := kubernetes.Client.AppsV1().Deployments("default").List(context.TODO(), metav1.ListOptions{})
+
+		if err != nil {
+			return err
+		}
+
+		for _, deployment := range deployments.Items {
+			if strings.Contains(deployment.Name, "miner") || strings.Contains(deployment.Name, "poet") {
+				err := kubernetes.Client.AppsV1().Deployments("default").Delete(context.TODO(), deployment.Name, metav1.DeleteOptions{})
+
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		err = kubernetes.Client.CoreV1().Namespaces().Delete(context.TODO(), "ws", metav1.DeleteOptions{})
```

Comment:
Are you sure that this finalizes everything we need to delete (which doesn't logs/metrics)?

Reply:
Yes I have tested this.